src/main/java/at/hannibal2/skyhanni/features/fishing/FishingTimer.kt

```diff
@@ -3,126 +3,220 @@ package at.hannibal2.skyhanni.features.fishing
+        if (babyMagmaSlugsToFind == 0) return
+        val location = lastMagmaSlugLocation ?: return
+        val slugs = recentBabyMagmaSlugs.filter { it !in mobMap }
+            .sortedBy { it.baseEntity.distanceTo(location) }
+            .take(babyMagmaSlugsToFind).ifEmpty { return }
+        babyMagmaSlugsToFind -= slugs.size
+        slugs.forEach { mobMap[it] = SimpleTimeMark.now() }
+        if (babyMagmaSlugsToFind == 0) {
+            recentBabyMagmaSlugs.clear()
+            lastMagmaSlugLocation = null
         }
+        updateInfo()
+    }
 
-        if (inHollows && newCount >= 60 && config.wormLimitAlert) {
-            SoundUtils.playBeepSound()
-            LorenzUtils.sendTitle("§cWORM CAP FULL!!!", 2.seconds)
+    @SubscribeEvent
+    fun onKeyPress(event: LorenzKeyPressEvent) {
+        if (!isEnabled()) return
+        if (Minecraft.getMinecraft().currentScreen != null) return
+        if (config.manualResetTimer.isKeyClicked()) {
+            mobMap.replaceAll { _, _ ->
+                SimpleTimeMark.now()
+            }
         }
     }
 
-    private fun countMobs() =
-        EntityUtils.getEntities<EntityArmorStand>().map { entity -> FishingAPI.seaCreatureCount(entity) }.sum()
-
-    private fun isRightLocation(): Boolean {
-        inHollows = false
-
-        if (config.forStranded && LorenzUtils.isStrandedProfile) return true
-
-        if (config.crystalHollows && IslandType.CRYSTAL_HOLLOWS.isInIsland()) {
-            inHollows = true
-            return true
+    private fun updateInfo() {
+        currentCount = mobMap.entries.sumOf {
+            1 + it.key.extraEntities.size
         }
+        startTime = mobMap.maxByOrNull { it.value.passedSince() }?.value ?: SimpleTimeMark.farPast()
+    }
 
-        if (config.crimsonIsle && IslandType.CRIMSON_ISLE.isInIsland()) return true
-
-        if (config.winterIsland && IslandType.WINTER.isInIsland()) return true
-
-        if (!IslandType.THE_FARMING_ISLANDS.isInIsland()) {
-            return LocationUtils.playerLocation().distance(barnLocation) < 50
+    private fun updateLocation() {
+        rightLocation = when (LorenzUtils.skyBlockIsland) {
+            IslandType.CRYSTAL_HOLLOWS -> config.crystalHollows.get()
+            IslandType.CRIMSON_ISLE -> config.crimsonIsle.get()
+            IslandType.WINTER -> config.winterIsland.get()
+            IslandType.HUB -> barnLocation.distanceToPlayer() < 50
+            IslandType.PRIVATE_ISLAND -> config.forStranded.get() && LorenzUtils.isStrandedProfile
+            else -> false
         }
-
-        return false
     }
 
     @SubscribeEvent
     fun onRenderOverlay(event: GuiRenderEvent.GuiOverlayRenderEvent) {
-        if (!LorenzUtils.inSkyBlock) return
-        if (!config.enabled) return
+        if (!isEnabled()) return
         if (!rightLocation) return
         if (currentCount == 0) return
         if (!FishingAPI.isFishing()) return
 
+        val text = createDisplay()
```

Comment:
Don't create the text each frame. rather only 10 times a second and cache the display string.